Support: Update support URL to relative path

jeremeylduvall opened the pull request:
This is an anticipated fix for #4320. It updates the URL in [`client/lib/url/support.js`](https://github.com/Automattic/wp-calypso/blob/master/client/lib/url/support.js#L15) from `https://wordpress.com/help/contact` to `/help/contact` so the URL works in all environments.

I also updated the reference we're using for "Contact" in two other areas:
-  [`purchases/list/index.jsx`](https://github.com/Automattic/wp-calypso/blob/4810232ff5c7e85accaddd8888a335a0c8ee99be/client/me/purchases/list/index.jsx#L37) when there's a cancellation problem. You can test this when running the branch by visiting `purchases/cancel-problem` and viewing the link in "Please contact support for more information."
- [`me/security-2fa-setup-backup-codes/index.jsx`](https://github.com/Automattic/wp-calypso/blob/8aa30953c0aec9647b59f05984967c5b337fad28/client/me/security-2fa-setup-backup-codes/index.jsx#L71) when there's an error obtaining backup codes. **I honestly haven't found a way to test this**, but I can't see why it couldn't work.

To make sure this was working correctly, I found a few cases where it's used and gave it a go:

First, [`client/components/happiness-support/index.jsx`](https://github.com/Automattic/wp-calypso/blob/96c5d26189b74a9f1bfcb7568f22ce3215e160ce/client/components/happiness-support/index.jsx#L31), which loads on the "Thank You" page, for example.

![screen shot 2016-03-29 at 4 06 10 pm](https://cloud.githubusercontent.com/assets/7240478/14125390/311ba5a4-f5c8-11e5-8f8d-9cbb751b70d8.png)

Second, [`domain-management/edit-contact-info/privacy-enabled-card.jsx`](https://github.com/Automattic/wp-calypso/blob/b229c821d3f17064e32487b2ada2d01dabbcceca/client/my-sites/upgrades/domain-management/edit-contact-info/privacy-enabled-card.jsx), which loads when users want to change contact information on a domain that has privacy active.

![screen shot 2016-03-29 at 16 07 52](https://cloud.githubusercontent.com/assets/7240478/14125433/6c16d570-f5c8-11e5-8405-3228ad7cadb4.png)

Finally, I loaded http://calypso.localhost:3000/purchases/cancel-problem with the branch running. The "contact support" link was correctly updated.

All three appear to work correctly. The easiest way to test this though is by checking out the App Components:

1. Run this branch locally
2. Visit the App Components here: http://calypso.localhost:3000/devdocs/app-components
3. View the link for "Ask a Question" under the `HappinessSupport` component

jordwest commented on client/lib/url/support.js:
```diff
@@ -12,7 +12,7 @@ export default {
 	COMMENTS: `${root}/category/comments`,
 	COMMUNITY_TRANSLATOR: `${root}/community-translator`,
-	CONTACT: `https://wordpress.com/help/contact`,
+	CONTACT: `/help/contact`,
```
Wondering if we should name this `CALYPSO_CONTACT` or `CALYPSO_CONTACT_SUPPORT` to avoid confusion with `JETPACK_CONTACT_SUPPORT` further down?

jeremeylduvall replied:
Probably a solid idea. I can rename it and then update the places we're currently using it.

jeremeylduvall commented:
Dug through and created a list of each place support.CONTACT (or some variation) is used so I can update it to support.CALYPSO_CONTACT once I make the change. Will tackle and test this today.

- [ ] wp-calypso/client/my-sites/upgrades/domain-management/edit-contact-info/privacy-enabled-card.jsx
- [ ] wp-calypso/client/components/happiness-support/index.jsx
- [ ] wp-calypso/client/my-sites/upgrades/domain-management/email-forwarding/email-forwarding-item.jsx
- [ ] wp-calypso/client/my-sites/upgrades/domain-management/transfer/unlocked.jsx
- [ ] wp-calypso/client/my-sites/upgrades/domain-management/transfer/shared.js
- [ ] wp-calypso/client/my-sites/upgrades/domain-management/email-forwarding/email-forwarding-add-new.jsx
- [ ] wp-calypso/client/me/purchases/cancel-purchase/index.jsx
- [ ] wp-calypso/client/my-sites/exporter/exporter.jsx
- [ ] wp-calypso/client/me/purchases/list/index.jsx
- [ ] wp-calypso/client/me/security-2fa-setup-backup-codes/index.jsx
- [ ] wp-calypso/client/my-sites/site-settings/delete-site/index.jsx

jeremeylduvall commented:
@jordwest mind taking another look sometime this week? 

Updated the reference from `CONTACT` to `CALYPSO_CONTACT`. Updated the corresponding references throughout Calypso to `support.CALYPSO_CONTACT`. I also updated a few instances where we were importing `supportUrls` vs. `support` for consistency. I was only able to test the cases originally mentioned since a lot of these instances require something to break in order for the link to show.

jordwest commented:
Nice! :+1: 
There's just a couple more cases where `CONTACT` is used:

 - `/client/me/purchases/manage-purchase/index.jsx` : 129
 - `/client/my-sites/upgrades/domain-management/transfer/shared.js`: 52

jeremeylduvall commented:
Well shoot. Not sure how I missed that. Updated both instances!

jordwest commented:
LGTM :+1:
